planning/autoware_planning_validator/src/utils.cpp

```diff
@@ -76,6 +76,45 @@ Trajectory resampleTrajectory(const Trajectory & trajectory, const double min_in
   return resampled;
 }
 
+double calculateStoppingDistance(
+  const double current_vel, const double current_accel, const double decel, const double jerk_limit)
+{
+  // calculate time to ramp acceleration from current accel to decel
+  const auto t1 = std::max((current_accel - decel) / jerk_limit, 0.0);
+  // calculate velocity and distance after t1
+  const auto v1 = current_vel + current_accel * t1 - 0.5 * jerk_limit * t1 * t1;
+  const auto d1 =
+    (current_vel * t1) + (0.5 * current_accel * t1 * t1) - (jerk_limit * t1 * t1 * t1 / 6.0);
+  // calculate distance to stop from v1
+  const auto d2 = std::abs((v1 * v1) / (2 * decel));
+  return d1 + d2;
+}
```

Comment:
@mkquda You can use this util function.
https://github.com/autowarefoundation/autoware_core/blob/main/common/autoware_motion_utils/src/distance/distance.cpp